### ISSUE TITLE
What are the "promotion" rules for cleaner files?

### ISSUE THREAD

**Issue by rbrito** [az0/cleanerml#95]
I have contributed some files to this project in the past, but I am simply not sure how/where I should be contributing some future work.

For instance, I see that the cleaning rules are divided into, essentially, three pieces:

* This project's `pending` directory.
* This project's `release` directory.
* Bleachbit's shipped rules.

I am running a new Firefox version (either aurora or, sometimes, even nightly) and there is a considerable amount of things that bleachbit 1.8 from Debian doesn't clean and I have to add my own personal rules.

Oh, and depending on the extensions to Firefox that you may have, you may gather a lot of garbage. But Firefox is only one of the many programs that need a face lift here.

Of course, I would like to give them back to the community (and let the community improve them while I am sleeping :)), but, really, I don't know where to send the patches!

In other words, this project (and bleachbit also) needs a `CONTRIBUTING.md` file. I can send an appropriate patch *if* I know how the project is structured.

Regards,

Rogério Brito.

**Comment by nodiscc** [comment]
Hi, the pending/release split was decided a long time ago when this repo was set up. In short, cleaners in `release/` have actually been tested over a long period (preferably by multiple users) and can be considered safe to run. I think @az0 planned to ship the `release/*` cleaners alongside the official bleachbit release but I don't know if this still stands.

If you want to add cleaner rules, you could add a new cleaner file to `pending/*` - I'll help with the review. Or we could place you changes in an experimental.

**Comment by az0** [comment]
This is a good question and deserves better clarification, and nodiscc is basically correct. 

* pending: The best place for new submissions. Subject to only light review.
* release: Tested by time and by users.
* BleachBit: Heavily tested. Used by many users. (Some there now have not been maintained or are used much and could be demoted.)

I hope users will indicate testing through GitHub comments or pull requests. The PR could be a comment in the XML like "tested with application version 123.456 on operating system foo 2015.4"

In the future I hope to integrate "release" and maybe "pending" into BleachBit with over-the-air updates similar to Winapp2.ini.

If this is not enough description to write a contributing file, please reopen.  Thank you!
